src/Types.hs

```diff
@@ -5,25 +5,45 @@ module Types where
 
 import           Control.Monad.IO.Class (MonadIO, liftIO)
 import           Control.Monad.Reader (MonadIO, MonadReader, ReaderT)
+import           Data.Aeson (FromJSON(parseJSON), Value(..), object, (.=), (.:), (.:?))
 import           Data.Configurator.Types
-import           Data.Aeson (FromJSON(parseJSON), Value(..), object, (.=), (.:))
 import           Web.Scotty.Trans (ActionT)
 import qualified Data.Text.Lazy as L
 
+data HayooResult = HayooResult { fname :: String, ftype :: Maybe String, flocation :: String } deriving (Show)
+
+instance FromJSON HayooResult where
+  parseJSON (Object v) = HayooResult <$> v .: "resultName" <*> v .:? "resultSignature" <*> v .: "resultUri"
```

Comment:
This is partial (only handles `Object`), are you using `-Wall`? `withObject` is good for this.